py27googledriveconnect/googledriveutils.py

```diff
@@ -0,0 +1,164 @@
+    no_folders_here = True  # We'll use this to decide whether we need to raise an error later
+    for afile in file_list:
+        if afile['mimeType'] == 'application/vnd.google-apps.folder':  # if file is a folder...
+            no_folders_here = False  # We won't need to raise the error
+            if afile['title'] == folder_name:
+                fid = afile['id'] # Look for folder that matches name, and save the folder ID
+                return fid
+            else:  # If nothing matched, the name was wrong
+                raise NotMatching('No folder of that name in specified directory')
+    if no_folders_here:  # if none of files in the list were folders, then say so.
+        raise NoFolders('There are no folders in specified directory')
+
+
+def find_reactorfolder(reactorno):
+    """
+    Finds the directory of a specified reactor. (reactors are numbered 1 thru 6)
+    :param reactorno: int, this is the reactor in question
+    :return: str, this is the file ID of the specified reactor's directory
+    :exception No Such Reactor: if no directory of the specified reactor exists, raise an error.
+    """
+    r_folder_name = 'R' + str(reactorno)  # folder will be 'R1', 'R2', etc. So make that string.
+    # Navigate through the directories
+    wlab_fid = find_folderid('Winkler Lab', 'root')
+    rdata_fid = find_folderid('ReactorData', wlab_fid)
+    try:  # When we find reactordata directory, look for our reactor's folder
+        r_fid = find_folderid(r_folder_name,rdata_fid)
+        return r_fid
+    except: # If we can't find it, say so.
+        raise NoSuchReactor('There is no reactor with that number')
+
+
+def find_reactorfileid(reactorno, filename):
+    """
+    Find a specific file withn a specified reactor's directory
+    :param reactorno: int, the number of reactor in question
+    :param filename: str, the name of file being looked for
+    :return: file, this is the file
+     OR
+    :return: boolean, this is false if there is no file matching specified name.
+                      The boolean is then used in write_to_reactordrive to make a new file
+    """
+    tgt_folder_id = find_reactorfolder(reactorno)  # Get file id of our reactor's folder
+    file_list = get_file_list(tgt_folder_id) # List all files in that folder
+    no_file = True
+    for afile in file_list:
+        if afile['title'] == filename:  # Look for file that matches name
+            no_file = False # Now we know that file exists, so we don't need to make it later
+            fileid = afile
+            return fileid # Return our file of interest
+    if no_file: # If the file we asked for doesn't exist, tell next function with a boolean.
+        return False
+
+
+def write_to_reactordrive(reactorno, filename, text):
+    """
+    Writes some specified info to a specified file for a specified reactor
+    :param reactorno: int, this is the reactor in question
```

Comment:
Can you elaborate?   when you say "Wouldn't it be better to specify 'int' type at the very end in parenthesis?" do you mean specify it last in the function definition, so it would be:
def read_from_reactordrive(filename, save_file_as,reactorno):

if so why? I did not know that mattered